### PR TITLE
USWDS-Site - Header: Add variants table to guidance page

### LINKED ISSUE #1976: USWDS-Site - Bug: Header has empty implementation sections
### Describe the bug

Header and its multiple variants have empty implementation, or "Using the header" sections.

The current logic to display this header and section exists in `component-guidance.html`

```html
  {% if implementation.size > guidanceNullLength or page.variants or site.data.settings.components[settingsKey] %}
```

### Steps to reproduce the bug

1. Visit the [header component page](https://designsystem.digital.gov/components/header/)
2. Scroll down to "Using the..." sections for all header variants
3. Confirm there is no content in that section

### Expected Behavior

We should add logic to remove this section if it's empty

### Screenshots

Screenshot 2023-01-18 at 3 51 18 PM
Screenshot 2023-01-18 at 3 51 36 PM

### System setup

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate bug reports.

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

**Added header variants table to guidance page.** This change also removes empty "**Using the header**" section.

## Related issue

Closes #1976 

## Preview link

Preview link:
[Header variants →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-header-variant-guidance/components/header/#header-variants)
[Header changelog →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-header-variant-guidance/components/header/#latest-updates)

## Problem statement

The Header guidance page was missing a variants table that can be found in other component guidance pages. There were also empty implementation sections due to each component containing variants in its frontmatter, which were not displayed in the sections.

## Solution

Move the `variants` out of the individual header variants front matter and include it in the base header guidance markdown page.

## Testing and review

1. Visit [Header guidance page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-header-variant-guidance/components/header/)
2. View variants table and check for missing variants
3. Check each variant section and check for empty implementation sections _(screenshots below)_

### Empty implementation section
_There's an empty **Using the...** section for each variant_
image
[As seen on live site →](https://designsystem.digital.gov/components/header/#:~:text=of%20a%20megamenu.-,Using%20the%20basic%20header,-Package)

### New variants table
image

Before opening this PR, make sure you’ve done whichever of these applies to you:
- [x] Confirm that this code follows the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `git pull origin [base branch]` to pull in the most recent updates from your base and check for merge conflicts. (Often, the base branch is `main`).
- [x] Run `npm test` and confirm that all tests pass.
- [x] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.

### CODE REVIEW

File: _data/changelogs/component-header.yml [hunk 1]
```diff
@@ -2,6 +2,12 @@ title: Header
 type: component
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Added Header variants table to guidance page.
```
Comment (amyleadem) [inline _data/changelogs/component-header.yml:6]:
```suggestion
    summary: Added variants table to guidance page.
```

Comment (amyleadem) [inline _data/changelogs/component-header.yml:6]: We should not capitalize the component name.

```suggestion
    summary: Added header variants table to guidance page.
```

File: _data/changelogs/component-header.yml [hunk 2]
```diff
@@ -2,6 +2,12 @@ title: Header
...
+    affectsGuidance: true
+    githubPr: 2139
+    githubRepo: uswds-site
+    versionUswds: 3.6.0
```
Comment (amyleadem) [inline _data/changelogs/component-header.yml:10]: No need to add a uswds version if the change is not tied to uswds code updates.
```suggestion
```

**Review by amyleadem — CHANGES_REQUESTED** [review]
Thanks for finding and fixing this, @mahoneycm! Overall it looks good, I just added a couple suggestions for small tweaks in the changelog. 

- [x] Confirmed listed variants match those found in USWDS
- [x] Confirmed the empty implementation sections have been removed
- [x] Proofed for grammar and spelling

**Comment by mahoneycm** [comment]
@mejiaj Bumping for review

**Review by amyleadem — APPROVED** [review]
Looks good to me!

Just flagging that we should update the changelog date before merge.

**Review by mejiaj — CHANGES_REQUESTED** [review]
Looks good. Confirming:

- [x] Variants table exists and is accurate.
- [x] Empty sections are gone.
- [x] Changelog entry exists.

@mahoneycm would you be able to briefly look into:

1. A visual regression with touchpoints. If confirmed, can you create an issue if there isn't one already?
2. Updating changelog date to today, so we can get this merged.

**Preview** (left) / **Live** (right)
image

**Comment by mahoneycm** [comment]
@mejiaj Date updated and issue created, chef! 🫡

https://github.com/uswds/uswds-site/issues/2186

**Review by mejiaj — APPROVED** [review]
Thanks! I've pulled in latest from `main` branch.

**Pull request merged**